graal-js/src/com.oracle.truffle.js.test/src/com/oracle/truffle/js/test/interop/ESModuleTest.java

```diff
@@ -211,6 +215,356 @@ private static void deleteFiles(File[] filesArray) {
         }
     }
 
+    /**
+     * Test basic module block features via mimicking the {@code} testFunctionExport() +
+     */
+    @Test
+    public void testModuleBlockMimicFunctionExport() throws IOException {
+        File[] allFilesArray = null;
+
+        TestOutput out = new TestOutput();
+        try (Context context = JSTest.newContextBuilder().allowHostAccess(HostAccess.ALL).allowIO(true).err(out).out(out).option(
+                        JSContextOptions.CONSOLE_NAME, "true").option(JSContextOptions.UNHANDLED_REJECTIONS_NAME, "throw").build()) {
+            allFilesArray = prepareTestFileAndModules("resources/moduleBlock/moduleBlockFunctionExportTestMimic.js",
+                            "resources" + "/functionexportmodule.js");
+
+            Source mainSource = Source.newBuilder(ID, allFilesArray[0]).mimeType("application/javascript+module").build();
+
+            Value v = context.eval(mainSource);
+
+            System.out.println(out);
```

Comment:
Not an expert on this test suite, but I find it weird to `print` during a test. Do you read that anywhere and compare to an expected output? If not, you should avoid printing (except for debug output during development) and rather use JUnit's `Assert.assertTrue` etc.

Reply:
Turned the print into assertions.